sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -149,9 +154,17 @@ def _buildprovider(
     # pylint:disable=protected-access
     provider = AzureAppConfigurationProvider()
     headers = kwargs.pop("headers", {})
-    headers["Correlation-Context"] = _get_correlation_context(key_vault_options)
+    if os.environ.get(REQUEST_TRACING_DISABLED_ENVIRONMENT_VARIABLE, default="").lower() != "true":
```

Comment:
Is there a way to extract all this common code to a utility somewhere? Its duplicated code and we might forget to update in in one place if more changes are needed in future